### PR TITLE
git-gud: allow decimal point in email

### PR DESCRIPTION
**Pull request by bravl**
Update regex to allow a decimal point in the email string.

### CODE REVIEW

File: git-gud [hunk 1]
```diff
@@ -6,7 +6,7 @@ validateName() {
   [[ "$1" =~ $re ]]
 }
 validateEmail() {
-  local re="^[a-zA-Z0-9]+@[a-zA-Z0-9]+\.[a-z]{2,}$"
+  local re="^[a-zA-Z0-9\.]+@[a-zA-Z0-9]+\.[a-z]{2,}$"
```
Comment (aaossa) [inline git-gud]: Regexes are so complex... Seems like dots are allowed in the local part of an email if there are no two consecutive dots and also no leading or trailing dots 🤔 Can you check if this work for you?

`^(?!.*\.\.)(?!.*\.$)\w*@[a-zA-Z0-9]+\.[a-z]{2,}$`

Comment (bravl) [reply git-gud]: That one did not work, `^(?!.*\.\.)^[a-zA-Z0-9][a-zA-Z0-9\.]*@[a-zA-Z0-9]+\.[a-z]{2,}$` seems to work with the cornercases you've described.

image

Hmm, golang and python seem happy with that regex but not `sh`..

Comment (bravl) [reply git-gud]: Fixed.

Comment (aaossa) [reply git-gud]: Seems like a trailing dot is still allowed but shouldn't (`something.@something.com` must be rejected). Maybe `^(?!.*\.\.)^[a-zA-Z0-9\.]*[a-zA-Z0-9]@[a-zA-Z0-9]+\.[a-z]{2,}$` instead?

Comment (aaossa) [reply git-gud]: Didn't notice that the regex in your PR already addressees this, and it seems to be ok so let's accept that 👌

**Comment by bravl** [comment]
Thanks!

**Pull request merged**